Server/src/services/authenticationService.ts

```diff
@@ -7,214 +7,232 @@ import mailgun from 'mailgun-js';
-        if (!verifiedEmail) {
-            throw new BadRequest("If a user with such email address exists, you would receive an email");
-        }
-        else {
-            let jwtParams: IJwtParams;
-            try {
-                jwtParams = await AuthenticationModel.obtainJWTParams(resetPasswordInformation.email);
-                accessToken = await this.generateJwtToken(jwtParams);
-                resetTokenChanged = await AuthenticationModel.resetPassword(resetPasswordInformation.email, accessToken);
-            } catch (error) {
-                throw new InternalServerError("Internal server error", error.message);
-            }
-
-            if (!resetTokenChanged) {
-                throw new InternalServerError("Internal server error when executing set resetToken on user");
-            }
-
-            let data = {
-                from: "noreply@databoom.com",
-                to: resetPasswordInformation.email,
-                subject: 'Reset your password',
-                html: `
-                    <h2>To reset your password, please click on this link</h2>
-                    <p>${process.env.CLIENT_URL}/resetpassword/${accessToken}</p>
-                `
-            };
-
-            mg.message().send(data, function (error, body) {
-                if (error) {
-                    this.requestResponse.statusCode = 400;
-                    this.requestResponse.message = error.message;
-                } else {
-                    this.requestResponse.statusCode = 200;
-                    this.requestResponse.message = "Email has been sent";
-                }
-            });
-        }
-        return this.requestResponse;
+    if (resetPasswordInformation.password === undefined) {
+      throw new BadRequest("The token has expired");
+    }
+    else {
+      try {
+        let passwordHash = await this.hashPassword(resetPasswordInformation.password);
+        await AuthenticationModel.resetPasswordByToken(resetToken, passwordHash);
+        this.requestResponse.message = "Password reset successfully"
+        this.requestResponse.statusCode = 200
+      } catch (error) {
+        throw new InternalServerError(error.message);
+      }
     }
+    return this.requestResponse;
+  }
 
-    /**
-     * Resposible for hashing the user input password
-     * @param password - user password
-     */
-    async hashPassword(userPassword: string): Promise<string> {
+  async forgotPassword(forgotPasswordInfo: IForgotPasswordInformation): Promise<IResponse> {
+    let verifiedEmail: boolean;
+    let accessToken: string;
+    let resetTokenChanged: boolean;
 
-        let hashedPassword: string;
-        hashedPassword = await argon2.hash(userPassword);
-        return hashedPassword;
+    verifiedEmail = await AuthenticationModel.verifyIfEmailExists(forgotPasswordInfo.email);
+    if (!verifiedEmail) {
+      throw new BadRequest("If a user with such email address exists, you would receive an email");
```

Comment:
This error message is confusing; it implies an email can/would be sent despite this catch stopping that.

Reply:
No, this is a way so whoever is using the service won't be able to find out which emails are registered and which are not